Design/ISIS_Imaging/ISIS_Imaging_general.md

```diff
@@ -0,0 +1,182 @@
+<!-- TOC -->
+
+- [Implementation design for ISIS_Imaging GUI](#implementation-design-for-isis_imaging-gui)
+- [Motivation](#motivation)
+- [Requirements](#requirements)
+- [Design](#design)
+    - [General structure](#general-structure)
+    - [Loading](#loading)
+    - [Handling multiple stacks in dialogues](#handling-multiple-stacks-in-dialogues)
+    - [Visualisation](#visualisation)
+    - [Applying a filter](#applying-a-filter)
+    - [Undoing an operation](#undoing-an-operation)
+    - [Histograms](#histograms)
+    - [Processing a full volume](#processing-a-full-volume)
+    - [Issues with Finding Center of Rotation and Reconstruction](#issues-with-finding-center-of-rotation-and-reconstruction)
+        - [Automatic Center of Rotation (COR) with imopr cor](#automatic-center-of-rotation-cor-with-imopr-cor)
+        - [Center of Rotation (COR) with imopr corwrite](#center-of-rotation-cor-with-imopr-corwrite)
+    - [Reconstruction](#reconstruction)
+    - [Remote submission](#remote-submission)
+        - [Remote compute resource used at ISIS: SCARF](#remote-compute-resource-used-at-isis-scarf)
+
+<!-- /TOC -->
+
+# Implementation design for ISIS_Imaging GUI
+The user requirements for the interface have been listed [here](https://github.com/mantidproject/isis_imaging/wiki/High-Level-User-Requirements-and-Use-Cases) and have been reviewed and confirmed by the imaging scientists.
+
+The development requirements, following from the user requirements, are listed [here](https://github.com/mantidproject/isis_imaging/wiki/High-Level-Development-Requirements-and-Guidelines)
+
+# Motivation
+The current [Tomography Reconstruction](https://github.com/mantidproject/mantid/tree/043095a619bc8851942a31253a1a8f8b820ab30f/MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/Tomography) interface was found to not satisfy the current imaging requirements. An easy solution to the problems was not found. With the agreement of the IMAT imaging scientists work is to be started on this design for a rewrite of the interface. Python is the language of choice, in order to be able to make use of visualising libraries like Matplotlib and reconstruction packages like Tomopy.
```

Comment:
With agreement from IMAT imaging scientists and senior members of the Mantid team.